### ISSUE TITLE
[Feature Request] Pick up app name from server instance

### ISSUE THREAD

**Issue by Batalex** [plotly/dash#771]
**Is your feature request related to a problem? Please describe.**
[It is known](https://community.plot.ly/t/dash-app-does-not-load-assets-and-app-index-string/12178/10?u=chriddyp) that the Dash app needs `name=__name__` to correctly load additional assets with gunicorn/waitress. However it is a shame that it cannot pickup the flask server's name if one is passed.
This leads to this:
```
server = Flask(__name__)
app = dash.Dash(server=server, name=__name__, external_stylesheets=external_stylesheets)
```

**Describe the solution you'd like**
Use flask server name if server in args but not name. See here https://github.com/plotly/dash/blob/master/dash/dash.py#L245

**Additional context**
I would be glad to submit a PR.

**Comment by alexcjohnson** [comment]
Great idea @Batalex! I don't see any problems with doing that, so yeah, we would welcome a PR 🎉

**Comment by alexcjohnson** [comment]
Closed by @Batalex in #774
